### PR TITLE
fix: remove prometheus/client_model Fixes #7815

### LINKED ISSUE #7815: Remove `github.com/prometheus/client_model` dependency
# Summary

The module is deprecated and is only used by tests.

---

**Message from the maintainers**:

Love this enhancement proposal? Give it a 👍. We prioritise the proposals with the most 👍.

### PR DESCRIPTION
**Pull request by prithvihv**
Fixes #7815: remove prometheus/client_model

Hi @blkperl, 
I would consider myself as intermediate at Golang + Kubernetes, I participated in this ticket to help me understand the flows and general structures present in this project. Would love to continue contributing

### CODE REVIEW

**Comment by prithvihv** [comment]
sorry, didn't mean to mark the PR to be ready for review

**Comment by prithvihv** [comment]
@blkperl in the PR description template, it mentions to run `--sign-off`, in the newer versions of git this flag has been changed to `--signoff`. Can consider changing this

**Comment by blkperl** [comment]
@prithvihv Do you mean the `sign-off` flag here? https://github.com/argoproj/argo-workflows/blob/master/.github/pull_request_template.md?plain=1#L19 

Would you like to open a PR to update that?

**Review by blkperl — COMMENTED** [review]
Upon closer inspection of the deprecation notice. I'm not sure we can drop the library yet as the go client still depends on it. The change in this PR removes test coverage that I think we want to keep. 

https://github.com/prometheus/client_model#deprecation-note


> The [Go instrumentation library](https://github.com/prometheus/client_golang), however, has been built around the protobuf-based data model. As a byproduct thereof, it is still able to expose the protobuf-based exposition format. The Go instrumentation library is the only remaining repository within the [Prometheus GitHub org](https://github.com/prometheus) directly using the prometheus/client_model repository.

**Comment by prithvihv** [comment]
@blkperl can we close the issue and this pr then?

File: go.mod [hunk 1]
```diff
@@ -36,7 +36,7 @@ require (
 	github.com/mitchellh/go-ps v0.0.0-20190716172923-621e5597135b
 	github.com/pkg/errors v0.9.1
 	github.com/prometheus/client_golang v1.12.1
-	github.com/prometheus/client_model v0.2.0
+	github.com/prometheus/client_model v0.2.0 // indirect
```
Comment (simster7) [inline go.mod:39]: Seems like library is still depended on regardless?

**Review by simster7 — CHANGES_REQUESTED** [review]
Not sure removing tests is an acceptable consequence of removing this library?

**Comment by blkperl** [comment]
@prithvihv yeah we can close both for now until the upstream library provides a path forward.

**Pull request closed without merging**